File: CS/Q236875/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using System.IO;
using DXSample;
using System.Collections;

namespace Q236875 {
    public partial class Form1 : XtraForm {
        LayoutSettingsPersistHelper layoutSettingsHelper;
        private const string layoutPath = @"..\..\layout.xml";
        public Form1() {
            InitializeComponent();
            List<ControlWithName> controls = new List<ControlWithName>();
            controls.Add(new ControlWithName(dockManager1, "dockManager1"));
            controls.Add(new ControlWithName(gridView1, "gridView1"));
            layoutSettingsHelper = new LayoutSettingsPersistHelper(controls.ToArray());
        }

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
            this.productsTableAdapter.Fill(this.nwindDataSet.Products);
        }

        private void OnSaveLayoutButtonClick(object sender, EventArgs e) {
            layoutSettingsHelper.SaveLayout(layoutPath);
            XtraMessageBox.Show("Complete!", "DXSample");
        }

        private void OnRestoreLayoutClick(object sender, EventArgs e) {
            layoutSettingsHelper.RestoreLayout(layoutPath);
            XtraMessageBox.Show("Complete!", "DXSample");
        }

        private void OnSaveDockLayoutButtonClick(object sender, EventArgs e) {
            layoutSettingsHelper.SaveLayout(layoutPath, dockManager1);
            XtraMessageBox.Show("Complete!", "DXSample");
        }

        private void OnRestoreDockLayoutButtonClick(object sender, EventArgs e) {
            layoutSettingsHelper.RestoreLayout(layoutPath, dockManager1);
            XtraMessageBox.Show("Complete!", "DXSample");
        }

        private void OnSaveGridLayoutButtonClick(object sender, EventArgs e) {
            layoutSettingsHelper.SaveLayout(layoutPath, gridView1);
            XtraMessageBox.Show("Complete!", "DXSample");
        }

        private void OnRestoreGridLayoutButtonClick(object sender, EventArgs e) {
            layoutSettingsHelper.RestoreLayout(layoutPath, gridView1);
            XtraMessageBox.Show("Complete!", "DXSample");
        }
    }
}

File: VB/Q236875/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports System.IO
Imports DXSample
Imports System.Collections

Namespace Q236875
	Partial Public Class Form1
		Inherits XtraForm
		Private layoutSettingsHelper As LayoutSettingsPersistHelper
		Private Const layoutPath As String = "..\..\layout.xml"
		Public Sub New()
			InitializeComponent()
			Dim controls As New List(Of ControlWithName)()
			controls.Add(New ControlWithName(dockManager1, "dockManager1"))
			controls.Add(New ControlWithName(gridView1, "gridView1"))
			layoutSettingsHelper = New LayoutSettingsPersistHelper(controls.ToArray())
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'nwindDataSet.Products' table. You can move, or remove it, as needed.
			Me.productsTableAdapter.Fill(Me.nwindDataSet.Products)
		End Sub

		Private Sub OnSaveLayoutButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			layoutSettingsHelper.SaveLayout(layoutPath)
			XtraMessageBox.Show("Complete!", "DXSample")
		End Sub

		Private Sub OnRestoreLayoutClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton2.Click
			layoutSettingsHelper.RestoreLayout(layoutPath)
			XtraMessageBox.Show("Complete!", "DXSample")
		End Sub

		Private Sub OnSaveDockLayoutButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton3.Click
			layoutSettingsHelper.SaveLayout(layoutPath, dockManager1)
			XtraMessageBox.Show("Complete!", "DXSample")
		End Sub

		Private Sub OnRestoreDockLayoutButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton4.Click
			layoutSettingsHelper.RestoreLayout(layoutPath, dockManager1)
			XtraMessageBox.Show("Complete!", "DXSample")
		End Sub

		Private Sub OnSaveGridLayoutButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton5.Click
			layoutSettingsHelper.SaveLayout(layoutPath, gridView1)
			XtraMessageBox.Show("Complete!", "DXSample")
		End Sub

		Private Sub OnRestoreGridLayoutButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton6.Click
			layoutSettingsHelper.RestoreLayout(layoutPath, gridView1)
			XtraMessageBox.Show("Complete!", "DXSample")
		End Sub
	End Class
End Namespace